web_external/js/views/body/MapPanel.js

```diff
@@ -115,6 +134,10 @@ minerva.views.MapPanel = minerva.View.extend({
             this.legendWidget[datasetId].remove(datasetId);
             delete this.legendWidget[datasetId];
         }
+        var layerIndex = $.inArray(layer, this.map.featureInfoWidget.layers);
```

Comment:
This breaks in this case:

  * refresh the page
  * add any non-wms dataset to the map
  * remove the non-wms dataset from the map

In this case, `this.map.featureInfoWidget` has not been created and the line above fails.

Can you create the featureInfoWidget in the MapPanel's init method?

It's looking like you had to go through some contortions to get this all working, and I doubt I understand them.  This might relate to how you wrote the featureInfoWidget.  We may need a longer discussion to revise.